docs/src/guide/configuration.md

```diff
@@ -26,5 +26,9 @@ in `manifests/metacontroller.yaml`):
 | `--events-qps` | Rate of events flowing per object (default - 1 event per 5 minutes, e.g. `--events-qps=0.0033`) |
 | `--events-burst` | Number of events allowed to send per object (default 25, e.g. `--events-burst=25`) |
 | `--pprof-address` | Enable pprof and bind to endpoint /debug/pprof, set to 0 to disable pprof serving (default 0, e.g. `--pprof-address=:6060`) |
+| `--leader-election` | Determines whether or not to use leader election when starting metacontroller (default `false`, e.g., `--leader-election`) |
+| `--leader-election-resource-lock` | Determines which resource lock to use for leader election (default `leases`, e.g., `--leader-election-resource-lock=configmapsleases`). See the client-go documentation [leaderelection/resourcelock](https://pkg.go.dev/k8s.io/client-go/tools/leaderelection/resourcelock#pkg-constants) for valid values. |
```

Comment:
@mjsmith1028 I see that in crt-runtime they are defaulting to `leases` (instead of `configmapleases`) - https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/leaderelection/leader_election.go#L61 - it seems they migrated from `configmapleases` to `leases`, so I think we should use/default what they use

Reply:
Thanks @grzesuav, I was using `leases` in the code but had documented the wrong value `configmapleases`. I would like your opinion on how we should default this value. Currently, the latest controller runtime tag is v0.11.1 which still has `configmapleases` https://github.com/kubernetes-sigs/controller-runtime/blob/v0.11.1/pkg/leaderelection/leader_election.go#L64. The latest in master that you linked has the default to `leases`. 

In this PR, I defaulted the value explicitly to `leases`:

```
	leaderElectionResourceLock = flag.String("leader-election-resource-lock", resourcelock.LeasesResourceLock, "Determines which resource lock to use for leader election")
```

Do you think I should explicitly use `leases` (` resourcelock.LeasesResourceLock`) as the default or if no value is provided then just let controller runtime default the value? i.e., I would omit the resource lock and controller runtime would use `configmapleases` in v0.11.1. Then, the next version it will default to `leases`.

Reply:
I think we can explicitly set default value as current default value in master (in ctrl-runtime) - so to `leases`. Ctrl-runtime is used as a library, metacontroller - as an application. Also, it wasn't present before so we do not have migration required etc